homeassistant/components/onvif/__init__.py

```diff
@@ -123,37 +121,23 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
     )
 
 
-async def _get_snapshot_auth(hass, device, entry):
+async def _get_snapshot_auth(device):
```

Comment:
```suggestion
async def _get_snapshot_auth(device):
    if not device.capabilities.snapshot:
        return HTTP_DIGEST_AUTHENTICATION
    if len(device.profiles) == 0:
        return HTTP_DIGEST_AUTHENTICATION
```

Reply:
no profiles will fail out of `async_setup_entry` before this gets hit, but can definitely add the snapshot check.